airbyte-integrations/connectors/source-harvest/source_harvest/availability_strategy.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import logging
+from typing import Dict
+
+import requests
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+from requests import HTTPError
+
+
+class HarvestAvailabilityStrategy(HttpAvailabilityStrategy):
+    """
+    This class is tested as part of test_source.check_connection
+    """
+
+    def reasons_for_unavailable_status_codes(
+        self, stream: Stream, logger: logging.Logger, source: Source, error: HTTPError
+    ) -> Dict[int, str]:
+        reasons_for_codes: Dict[int, str] = {
+            requests.codes.UNAUTHORIZED: "Please ensure your credentials are valid.",
+            requests.codes.FORBIDDEN: "This is most likely due to insufficient permissions on the credentials in use.",
+            requests.codes.NOT_FOUND: "Please ensure that your account ID is properly set. If it is the case and you are still seeing this error, please contact Airbyte support.",
```

Comment:
Is this an error we've encountered before, in that we know if we provide the wrong ID, it will give us a 404? 

If not I'd take this one out of the list in case we get 404 because the stream slice was messed up or something like that

Reply:
Like mentioned in this error message, if the account ID is not set properly, this is the HTTP status code that will be provided. Hence, it felt like it made sense to put this as a config error

Reply:
Ok! If I were only thinking about the platform workflow, I would wonder if we should put this exclusively in `check` and not in the availability strategy, because the value of the account ID in the config is not something you can change without going through check again. Compared to an account's permissions to access a certain resource, which could be revoked on the user's end. I guess a whole account could be revoked... 

But given that for other use cases (API etc) this could be the case that the account ID has changed in between, I'm fine leaving it. This is a wider problem though that made me try to start the conversation [here](https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1708723575728979).